ONSdigital/rm-sample-service
# fix sample load infinite loop

# What and why?
This PR is an attempt at fixing a bug where r-ops hangs when attempting to load a sample file with a duplicated sample unit. The code has changed so that an error message box appears on the page to indicate that there's a problem with the file.

# How to test?
* Create a sample file that has a duplicate sample unit ref.
* Deploy this PR to your environment and run acceptance tests.
* Create a collection exercise and attach the aforementioned sample file to the CE.
* Keep refreshing the page to check the sample file load process. At first, it will look like it's hanging, but after a while, it will eventually display an "error loading sample" box on the page.
* Check the logs for the sample service pod. There should be some stuff about the sample summary getting set to failed (which didn't happen before).

# Trello
[Card](https://trello.com/c/upvtJmqp)
